.github/actions/paths-filter/action.yml

```
@@ -280,11 +281,29 @@ runs:
           - optimize/c4/**
           - optimize/backend/src/main/resources/localization/**
 
-  - id: filter-stable-branch
+  # only works `on: push` and `on: pull_request` GitHub trigger events
+  - id: filter-special-branch
     shell: bash
     run: |
-      if [[ '${{ github.ref_name }}' =~ ^stable\/[0-9]+\.[0-9]+$ ]]; then
+      if [[ $GITHUB_EVENT_NAME == 'push' ]]; then
+        test_branch_name="$GITHUB_REF_NAME"
+      elif [[ $GITHUB_EVENT_NAME == 'pull_request' ]]; then
+        test_branch_name="$GITHUB_HEAD_REF"
+      fi
+      echo "Branch name: $test_branch_name"
+
+      if [[ $test_branch_name =~ ^stable\/[0-9]+\.[0-9]+$ ]]; then
+        echo "Stable branch: true"
         echo "stable-branch=true" >> "$GITHUB_OUTPUT"
       else
+        echo "Stable branch: false"
         echo "stable-branch=false" >> "$GITHUB_OUTPUT"
       fi
+
+      if [[ $test_branch_name =~ ^release.*$ ]]; then
```

🔧 

Shall we be more concrete?
```suggestion
      if [[ $test_branch_name =~ ^release\-.+$ ]]; then
```

---

In the past I saw cases of `release/*` being used, which is why in https://github.com/camunda/camunda/wiki/CI-&-Automation#git-branches I generalized it to `release*`.